common/scripts/mesh-helper.sh

```diff
@@ -43,21 +43,34 @@ EOF
   fi
 }
 
-generate_br_lan_ip() {
+generate_bridge_ip() {
   local mesh_if_mac
-  mesh_if_mac="$(ip -brief link | grep "$MESH_VIF" | awk '{print $3; exit}')"
+
+  bridge_name=$(echo "$BRIDGE" | cut -d' ' -f1)
+
+  mesh_if_mac="$(ip -brief link | grep "${id0_MESH_VIF}" | awk '{print $3; exit}')"
+  if [ -z "$mesh_if_mac" ]; then
+      echo "generate_bridge_ip: MAC not found for id0_MESH_VIF! Configuration error?" > /dev/kmsg
+      mesh_if_mac="$(ip -brief link | grep "eth0" | awk '{print $3; exit}')"
+  fi
   local ip_random
   ip_random="$(echo "$mesh_if_mac" | cut -b 16-17)"
-  br_lan_ip="192.168.1."$((16#$ip_random))
+  bridge_ip="192.168.1."$((16#$ip_random))
+
+  # legacy support
+  br_lan_ip=$bridge_ip
 }
 
 source_configuration() {
-  if [ -f "/opt/mesh.conf" ]; then
+  # $1: index of the mesh.conf file to be used
```

Comment:
If I may recommend as a coding style requirement or something: since shell script functions are not clear at all about which arguments they take if any, please ensure that at least the first few lines of the function body show them, for example:
`MESH_CONF_FILE_INDEX="$1"`
`WHAT_THE_2ND_ARG_IS="$2"`
`WHAT_THE_3RD_ARG_IS="$3"`
and so on :)

Reply:
This is not enough?
`# $1: index of the mesh.conf file to be used`

Reply:
Your comment line is not bad in itself, I saw it as the very first line of the function indeed, but later in the function it would still be better to have clear variable names than having to remember "What was $1 again?" 🙂

It would be also safer as it's much harder to typo $WHAT_I_MEANT into $SOMETHING_COMPLETELY_DIFFERENT than it is to typo $1 into $2

Reply:
Minor thing, but I will fix this.

Reply:
Fixed in https://github.com/tiiuae/mesh_com/pull/319/commits/d730ee831cee33375488ea0117601abd4c7f81da